internal/credit/postgres_connector/watermark.go

```diff
@@ -0,0 +1,44 @@
+package postgres_connector
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"entgo.io/ent/dialect/sql"
+
+	"github.com/openmeterio/openmeter/internal/credit/postgres_connector/ent/db"
+	db_credit "github.com/openmeterio/openmeter/internal/credit/postgres_connector/ent/db/creditentry"
+	credit_model "github.com/openmeterio/openmeter/pkg/credit"
+)
+
+// GetHighWatermark returns the high watermark for the given credit and subject pair.
+func (c *PostgresConnector) GetHighWatermark(ctx context.Context, namespace string, subject string) (credit_model.HighWatermark, error) {
+	lastReset, err := c.db.CreditEntry.Query().
```

Comment:
I would either add a limit 1 or use a `SELECT max() ...` query to help the query planner with making it explicit that we only need a single item.